editor/editor_node.cpp

```diff
@@ -1585,7 +1585,11 @@ void EditorNode::save_resource_in_path(const Ref<Resource> &p_resource, const St
 		return;
 	}
 
-	((Resource *)p_resource.ptr())->set_path(path);
+	Ref<Resource> prev_resource = ResourceCache::get_ref(p_path);
+	p_resource->set_path(path, true);
```

Comment:
You could do that inside the condition. No reason to set path if it's the same Resource.